cant nativeLink

[error] In file included from C:/projects/scala-native-test/target/scala-3.3.3/native/dependencies/windowslib_native0.5_3-0.5.1-3/scala-native/windows/time.c:10:
[error] C:/msys64/ucrt64/include/time.h:276:24: error: function cannot return array type 'char *[2]'
[error]   276 |   _CRTIMP extern char *tzname[2] __MINGW_ATTRIB_DEPRECATED_UCRT;
[error]       |                        ^
[error] C:/projects/scala-native-test/target/scala-3.3.3/native/dependencies/windowslib_native0.5_3-0.5.1-3/scala-native/windows/time.c:7:16: note: expanded from macro 'tzname'
[error]     7 | #define tzname _tzname
[error]       |                ^
[error] C:/msys64/ucrt64/include/time.h:126:26: note: expanded from macro '_tzname'
[error]   126 | #define _tzname (__tzname())
[error]       |                          ^
[error] C:/msys64/ucrt64/include/time.h:284:1: error: declaration of anonymous struct must be a definition
[error]   284 | struct timezone {
[error]       | ^
[error] C:/msys64/ucrt64/include/time.h:289:61: error: declaration of anonymous struct must be a definition
[error]   289 |   extern int __cdecl mingw_gettimeofday (struct timeval *p, struct timezone *z);
[error]       |                                                             ^
[error] C:/msys64/ucrt64/include/time.h:289:61: error: expected parameter declarator
[error] C:/projects/scala-native-test/target/scala-3.3.3/native/dependencies/windowslib_native0.5_3-0.5.1-3/scala-native/windows/time.c:122:38: warning: '__timezone' is deprecated: Only provided for source compatibility; this variable might not always be accurate when linking to UCRT. [-Wdeprecated-declarations]
[error]   122 | long scalanative_timezone() { return _timezone; }
[error]       |                                      ^
[error] C:/msys64/ucrt64/include/time.h:125:22: note: expanded from macro '_timezone'
[error]   125 | #define _timezone (* __timezone())
[error]       |                      ^
[error] C:/msys64/ucrt64/include/time.h:275:32: note: '__timezone' has been explicitly marked deprecated here
[error]   275 |   _CRTIMP extern long timezone __MINGW_ATTRIB_DEPRECATED_UCRT;
[error]       |                                ^
[error] C:/msys64/ucrt64/include/time.h:267:5: note: expanded from macro '__MINGW_ATTRIB_DEPRECATED_UCRT'
[error]   267 |     __MINGW_ATTRIB_DEPRECATED_MSG( \
[error]       |     ^
[error] C:/msys64/ucrt64/include/_mingw.h:189:58: note: expanded from macro '__MINGW_ATTRIB_DEPRECATED_MSG'
[error]   189 | #define __MINGW_ATTRIB_DEPRECATED_MSG(x) __attribute__ ((__deprecated__(x)))
[error]       |                                                          ^
[error] C:/projects/scala-native-test/target/scala-3.3.3/native/dependencies/windowslib_native0.5_3-0.5.1-3/scala-native/windows/time.c:124:37: warning: '__daylight' is deprecated: Only provided for source compatibility; this variable might not always be accurate when linking to UCRT. [-Wdeprecated-declarations]
[error]   124 | int scalanative_daylight() { return _daylight; }
[error]       |                                     ^
[error] C:/msys64/ucrt64/include/time.h:123:22: note: expanded from macro '_daylight'
[error]   123 | #define _daylight (* __daylight())
[error]       |                      ^
[error] C:/msys64/ucrt64/include/time.h:274:31: note: '__daylight' has been explicitly marked deprecated here
[error]   274 |   _CRTIMP extern int daylight __MINGW_ATTRIB_DEPRECATED_UCRT;
[error]       |                               ^
[error] C:/msys64/ucrt64/include/time.h:267:5: note: expanded from macro '__MINGW_ATTRIB_DEPRECATED_UCRT'
[error]   267 |     __MINGW_ATTRIB_DEPRECATED_MSG( \
[error]       |     ^
[error] C:/msys64/ucrt64/include/_mingw.h:189:58: note: expanded from macro '__MINGW_ATTRIB_DEPRECATED_MSG'
[error]   189 | #define __MINGW_ATTRIB_DEPRECATED_MSG(x) __attribute__ ((__deprecated__(x)))
[error]       |                                                          ^
[error] 2 warnings and 4 errors generated.
[info] Compiling to native code (9975 ms)
[info] Total (11475 ms)
[error] stack trace is suppressed; run 'last Compile / nativeLink' for the full output
[error] Failed to compile C:/projects/scala-native-test/target/scala-3.3.3/native/dependencies/windowslib_native0.5_3-0.5.1-3/scala-native/windows/time.c

--- a/windowslib/src/main/resources/scala-native/windows/time.c
+++ b/windowslib/src/main/resources/scala-native/windows/time.c
@@ -1,13 +1,39 @@
 // Windows specific copy of posixlib/time.h
 // Uses *_s variants of methods instead of *_r
 #if defined(_WIN32)
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+
+#if defined(_MSC_VER)
 #define _CRT_SECURE_NO_WARNINGS
 #define daylight _daylight
 #define timezone _timezone
 #define tzname _tzname
+void tzset() { _tzset(); }
+#else
+// — Non-MSVC Windows (MinGW, Cygwin, TDM-GCC, Clang-mingw, etc.) —
+// these CRTs already offer:
+//    asctime(), asctime_r(), gmtime_r(), localtime_r(), tzset()
+// so we just alias the “secure” names to the POSIX names
+// and provide asctime_s() if needed.
+#ifndef gmtime_s
+#define gmtime_s(dst, clk) gmtime_r((clk), (dst))
+#endif
 
-#include <string.h>
-#include <time.h>
+#ifndef localtime_s
+#define localtime_s(dst, clk) localtime_r((clk), (dst))
+#endif
+
+static errno_t asctime_s(char *buf, size_t size, const struct tm *tm) {
+    char *s = asctime(tm);
+    size_t len = strlen(s) + 1;
+    if (len > size)
+        return ERANGE;
+    memcpy(buf, s, len);
+    return 0;
+}
+#endif
 
 struct scalanative_tm {
     int tm_sec;
@@ -37,6 +63,7 @@ static void scalanative_tm_init(struct scalanative_tm *scala_tm,
 }
 
 static void tm_init(struct tm *tm, struct scalanative_tm *scala_tm) {
+    *tm = (struct tm){0}; // zero everything to avoid garbage
     tm->tm_sec = scala_tm->tm_sec;
     tm->tm_min = scala_tm->tm_min;
     tm->tm_hour = scala_tm->tm_hour;
@@ -118,12 +145,6 @@ size_t scalanative_strftime(char *buf, size_t maxsize, const char *format,
 }
 
 char **scalanative_tzname() { return tzname; }
-
-long scalanative_timezone() { return _timezone; }
-
-int scalanative_daylight() { return _daylight; }
-
-// Windows compat
-void tzset() { return _tzset(); }
-
+long scalanative_timezone() { return timezone; }
+int scalanative_daylight() { return daylight; }
 #endif // defined(_WIN32)

--- a/windowslib/src/main/resources/scala-native/windows/time.c
+++ b/windowslib/src/main/resources/scala-native/windows/time.c
@@ -1,12 +1,12 @@
 // Windows specific copy of posixlib/time.h
 // Uses *_s variants of methods instead of *_r
 #if defined(_WIN32)
+#define _CRT_SECURE_NO_WARNINGS
 #include <string.h>
 #include <time.h>
 #include <errno.h>
 
 #if defined(_MSC_VER)
-#define _CRT_SECURE_NO_WARNINGS
 #define daylight _daylight
 #define timezone _timezone
 #define tzname _tzname

